### PR TITLE
Upgrade to Gradle 4.6

### PR DESCRIPTION
**Pull request by garyrussell**
- add support for JUnit5 tests
- resume gradle testing of JUnit5 tests - `RabbitTemplateMPPIntegrationTests.java`, `SimpleMessageListenerContainerLongTests.java`

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -183,6 +184,10 @@ subprojects { subproject ->
 		}
 	}
 
+	testAll {
+		useJUnitPlatform()
```
Comment (artembilan) [inline build.gradle]: Looks like this can go to the next `tasks.withType(Test).all` block.

File: build.gradle [hunk 2]
```diff
@@ -89,9 +89,9 @@ subprojects { subproject ->
 		hamcrestVersion = '1.3'
 		jackson2Version = '2.9.1'
 		junit4Version = '4.12'
-		junitJupiterVersion = '5.0.2'
-		junitPlatformVersion = '1.0.2'
-		junitVintageVersion = '4.12.2'
+		junitJupiterVersion = '5.1.0'
+		junitPlatformVersion = '1.1.0'
```
Comment (artembilan) [inline build.gradle]: Is it OK to upgrade to the `minor` version in our point release? 
Junit is not only a test dependency. The Cairo platform must provide these versions as well.

What am I missing in our backward compatibility policy?

Thanks

Comment (garyrussell) [reply build.gradle]: Cairo gets its versions from [boot](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-dependencies/pom.xml#L114) so it seems we were wrong anyway.

Comment (garyrussell) [reply build.gradle]: Yes, it's a compile dependency for spring-rabbit-junit, but `optional` too; so I think we're ok.

**Comment by artembilan** [comment]
Good, but how about those two test classes. 
I still don't understand how your fix is related to them.

Thanks

**Comment by garyrussell** [comment]
Sorry, I missed your earlier comment.

Since they are JUnit5 tests, they have not been running in the gradle build. I was just commenting that they are now being tested again after this change is applied.

**Pull request merged**